rpc_server.go

```diff
@@ -334,6 +334,61 @@ func (s *SourceControl) WriteComment(comment *string, reply *bool) error {
 	return nil
 }
 
+// CouplingStatus describes the status of FB / error coupling
+type CouplingStatus int
+
+const (
+	NoCoupling CouplingStatus = iota + 1 // FB and error aren't coupled
+	FBToErr                              // FB triggers cause secondary triggers in error channels
+	ErrToFB                              // Error triggers cause secondary triggers in FB channels
+)
+
+// CoupleErrToFB turns on or off coupling of Error -> FB
+func (s *SourceControl) CoupleErrToFB(couple *bool, reply *bool) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.activeSource == nil {
+		return fmt.Errorf("No source is active")
+	} else if s.status.SourceName != "Lancero" {
+		return fmt.Errorf("CoupleErrToFB only works for Lancero sources")
+	}
+
+	*reply = true
+	c := NoCoupling
+	if *couple {
+		c = ErrToFB
+	}
+	err := s.activeSource.SetCoupling(c)
+	s.clientUpdates <- ClientUpdate{"TRIGCOUPLING", c}
+	if err != nil {
+		*reply = false
+	}
+	return err
+}
+
+// CoupleFBToErr turns on or off coupling of FB -> Error
+func (s *SourceControl) CoupleFBToErr(couple *bool, reply *bool) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.activeSource == nil {
+		return fmt.Errorf("No source is active")
+	} else if s.status.SourceName != "Lancero" {
```

Comment:
just make the AnySource method return an error instead of retruning nil, then you don't need to special case lancero here.